include/cuco/detail/static_multimap/device_view_impl.inl

```diff
@@ -1094,6 +1095,7 @@ class static_multimap<Key, Value, Scope, Allocator, ProbeSequence>::device_view_
         flush_output_buffer(g, *cg_counter, output_buffer, num_matches, output_begin);
         // First lane reset CG-level counter
         if (lane_id == 0) { *cg_counter = 0; }
+        g.sync();
```

Comment:
```suggestion
```
Seems redundant also since the read on line 1068 is right after a CG `ballot`. Similar to warp vote functions, CG vote functions like `ballot`, `any` and `all` imply an internal sync per [CUDA programming guide](https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#thread-block-tile).

Reply:
So `__ballot_sync` which is what the documentation for `g.ballot()` refers to in [warp vote functions](https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#warp-vote-functions) explicitly states

> These intrinsics do not imply a memory barrier. They do not guarantee any memory ordering.

So I _think_ that means without explicit synchronisation there is no happens-before ordering between the lane-zero write to `*cg_counter` and the read in the next loop iteration.

There's no problem I think between the write on line 1079 and the subsequent write on line 1097 because those are both done by the same thread.